x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings_by_resource/use_findings_by_resource.ts

```diff
@@ -22,77 +33,99 @@ export type CspFindingsByResourceResult = FindingsQueryResult<
-  size: 0,
   body: {
     query,
+    size: 0,
     aggs: {
-      groupBy: {
-        composite: {
-          size: 10 * 1000,
-          sources: [
-            { resource_id: { terms: { field: 'resource_id.keyword' } } },
-            { cluster_id: { terms: { field: 'cluster_id.keyword' } } },
-            { cis_section: { terms: { field: 'rule.section.keyword' } } },
-          ],
-        },
+      resource_total: { cardinality: { field: 'resource.id.keyword' } },
+      resources: {
+        terms: { field: 'resource.id.keyword', size: 1000000 },
         aggs: {
+          cis_sections: {
+            terms: { field: 'rule.section.keyword', size, order: { _key: 'asc' } },
+          },
+          name: {
+            terms: { field: 'resource.name.keyword', size: 1, order: { _key: 'asc' } },
+          },
+          type: {
+            terms: { field: 'resource.sub_type.keyword', size: 1, order: { _key: 'asc' } },
+          },
           failed_findings: {
             filter: { term: { 'result.evaluation.keyword': 'failed' } },
           },
+          sort_failed_findings: {
+            bucket_sort: {
+              from,
+              size,
+              sort: [
+                {
+                  'failed_findings>_count': { order: 'desc' },
+                  _count: { order: 'desc' },
+                  _key: { order: 'asc' },
+                },
+              ],
+            },
+          },
         },
       },
     },
   },
 });
 
-export const useFindingsByResource = ({ index, query }: FindingsBaseEsQuery) => {
+export const useFindingsByResource = ({ index, query, from, size }: UseResourceFindingsOptions) => {
   const {
     data,
     notifications: { toasts },
   } = useKibana().services;
 
   return useQuery(
-    ['csp_findings_resource', { index, query }],
+    ['csp_findings_resource', { index, query, size, from }],
     () =>
       lastValueFrom(
         data.search.search<FindingsAggRequest, FindingsAggResponse>({
-          params: getFindingsByResourceAggQuery({ index, query }),
+          params: getFindingsByResourceAggQuery({ index, query, from, size }),
         })
       ),
     {
+      keepPreviousData: true,
+      refetchOnWindowFocus: false,
       select: ({ rawResponse }) => ({
-        page: rawResponse.aggregations?.groupBy.buckets.map(createFindingsByResource) || [],
+        page: ((rawResponse.aggregations?.resources.buckets as FindingsAggBucket[]) || []).map(
+          createFindingsByResource
+        ),
+        total: rawResponse.aggregations?.resource_total.value,
       }),
       onError: (err) => showErrorToast(toasts, err),
     }
   );
 };
 
 const createFindingsByResource = (bucket: FindingsAggBucket) => ({
-  ...bucket.key,
+  resource_id: bucket.key,
+  cis_sections: (
+    bucket.cis_sections.buckets as estypes.AggregationsStringRareTermsBucketKeys[]
+  ).map((v) => v.key),
   failed_findings: {
-    total: bucket.failed_findings.doc_count,
+    total: bucket.doc_count,
```

Comment:
This is an error, it displays wrong number of failed findings, I believe this should be reverted to `bucket.failed_findings.doc_count`

Video:

https://user-images.githubusercontent.com/94899776/168770191-2b8b2fc1-2680-4d3c-a0d5-859459d4684a.mov